src/gentropy/dataset/study_locus.py

```diff
@@ -963,6 +965,76 @@ def qc_redundant_top_hits_from_PICS(self: StudyLocus) -> StudyLocus:
             _schema=StudyLocus.get_schema(),
         )
 
+    def qc_explained_by_SuSiE(self: StudyLocus) -> StudyLocus:
+        """Flag associations that are explained by SuSiE associations.
+
+        Credible sets overlapping in the same region as a SuSiE credible set are flagged as explained by SuSiE.
+
+        Returns:
+            StudyLocus: Updated study locus with SuSiE explained flags.
+        """
+        # unique study-regions covered by SuSie credible sets
+        susie_study_regions = (
+            self.filter(f.col("finemappingMethod") == "SuSiE-inf")
+            .df.select(
+                "studyId",
+                "chromosome",
+                "locusStart",
+                "locusEnd",
+                f.lit(True).alias("inSuSiE"),
+            )
+            .distinct()
+        )
+
+        # non SuSiE credible sets (studyLocusId) overlapping in any variant with SuSiE locus
+        redundant_study_locus = (
+            self.filter(~(f.col("finemappingMethod") == "SuSiE-inf"))
+            .df.withColumn("l", f.explode("locus"))
+            .select(
+                "studyLocusId",
+                "studyId",
+                "chromosome",
+                f.split(f.col("l.variantId"), "_")[1].alias("tag_position"),
+            )
+            .alias("study_locus")
+            .join(
+                susie_study_regions.alias("regions"),
+                how="inner",
+                on=[
+                    (f.col("study_locus.chromosome") == f.col("regions.chromosome"))
+                    & (f.col("study_locus.studyId") == f.col("regions.studyId"))
+                    & (f.col("study_locus.tag_position") >= f.col("regions.locusStart"))
+                    & (f.col("study_locus.tag_position") <= f.col("regions.locusEnd"))
+                ],
+            )
+            .select("studyLocusId", "inSuSiE")
+            .distinct()
+        )
+
+        return StudyLocus(
+            _df=(
+                self.df.join(redundant_study_locus, on="studyLocusId", how="left")
+                .withColumn(
+                    "qualityControls",
+                    self.update_quality_flag(
+                        f.col("qualityControls"),
+                        # credible set in SuSiE overlapping region
+                        f.col("inSuSiE")
+                        # credible set not based on SuSiE
+                        & ~(f.col("finemappingMethod") == "SuSiE-inf")
```

Comment:
I think this condition is not needed as the source dataset (`redundant_study_locus`) is already filtered.